### ISSUE TITLE
[RFC] ChatMessage - 'reactions' slot

### ISSUE THREAD

**Issue by mnajdova** [microsoft/fluent-ui-react#942]
### Problem description
We got request for adding additional slot to the ChatMessages for the reaction on the message.

### Teams reactions:
![image](https://user-images.githubusercontent.com/4512430/53176470-6be26b00-35ee-11e9-9f46-7503cc8e6b2a.png)

### Skype:
![image](https://user-images.githubusercontent.com/4512430/53176568-a51adb00-35ee-11e9-95c2-33a1d060461c.png)

### FB messanger:
![image](https://user-images.githubusercontent.com/4512430/53176941-68031880-35ef-11e9-8f39-c93f807188f9.png)

### Github post:
![image](https://user-images.githubusercontent.com/4512430/53176834-2a05f480-35ef-11e9-8fdb-dd0e95227dc1.png)

### Proposed solution
Seeing this is a common thing across messaging platforms, I would agree that it can be something added on the ChatMessage. The common thing between the reactions is that they have reaction (Icon maybe?) and number of reactions of that type. We can create ChatMessageReaction encapsulating this, and the ChatMessage may contain shorthand collection for reaction, which can be positioned on the top or bottom of the message.

So it would look something like

```jsx
<Chat.Message 
  author='John Doe' 
  timestamp='02/21' 
  content='Message with reactions' 
  reactions=[{icon: 'like', count: '15'}, {icon: 'angry', count: '1K'}] 
  reactionsPosition='top'
/>
```

I intentionally set the count as a string, because the number if big can be shorten - we may discuss this if it doesn't make sense in general. 

### MVP
[Fork, update, and replace this pen if you can show the proposed solution]:
https://codesandbox.io/s/2l3n74j9y

**Comment by bmdalex** [comment]
Looks good to me @mnajdova 

Few things:

### 1. Is `reactionsPosition` needed?

I am not sure about `reactionsPosition` prop; from your examples it looks like each theme will decide where to position the reactions; also it seems that it'd be pretty difficult to reason about:

- what possible values `reactionsPosition` will have; there can be so many: top, bottom, before, after, combinations of these, etc
- if a reaction has `reactionsPosition='bottom-after' how do we place that relative to the bottom right corner (above it or to the left of it) => **this indicates `reactionsPosition` might be a theme aspect**
- we don't want to enforce all theme to implement different values of `reactionsPosition` if they only need one (vast majority of cases)

### 2. Is `reactionsPosition` enough?

Also, if we add `reactionsPosition`, would that be enough? In some of these examples the reactions are positioned on the border or outside the chat container => **again, this indicates `reactionsPosition` might be a theme aspect**

**Comment by mnajdova** [comment]
Hey @Bugaa92 thanks for the feedback. The reason why I added the reactionsPosition was that, the HTML structure will be different if these needs to be added before or after the content. I agree that if we start with this we may open discussion for all different ways that it may be position, but at least allowing them to have the reactions in the correct place of the HTML structure and then use the theme to actually displayed it exactly where they need may be a good start...

**Comment by bmdalex** [comment]
@mnajdova good call, makes sense, thanks!

**Comment by kuzhelov** [comment]
let me share my thoughts on this proposal:
- provided examples look absolutely reasonable to me as proof to introduce corresponding slot to `ChatMessage`
- `ChatMessageReaction` - agree that we need to introduce dedicated component for that (I would omit naming at the moment and questions of whether it should be Chat-scoped component or general one, as I might expect that these 'reactions' may be seen as attribute of some other widget - but the fact that presentational logic should be encapsulated in dedicated component)
- as @Bugaa92, am concerned about positioning in general - while, at the same time, should explicitly mention that it shouldn't be a proposal's blocker, as this proposal just emphasizes the problem we've already had: with many slots provided to component it becomes hard to customize exact place of their appearance. 

So, speaking of the proposed plan to `positioning` - @mnajdova, could you, please, suggest what is the vision for this prop values, and, if possible, could you, please, also suggest JSX markup examples that would allow us to achieve views mentioned in the proposal (no need to introduce CSS styling part, it will be quite easy to imagine what it could be given the markup). Thank you very much!

**Comment by mnajdova** [comment]
@kuzhelov thanks for sharing your thoughts on this. 

- I would agree, maybe just Reaction as a separate component may be better, as this may be reuse on other places as well. 
- For the `reacitonsPosition` prop as I start was thinking of introducing the values `top` and `bottom` as for the ChatMessage, these should be enough for now (yes we can argue that we may even need as something `before` and `after`, but I am not sure whether we will need this). For starter, I agree that we may omit this and add it next to the timestamp slot (but in my opinion this is too specific to what we need at this moment for teams theme). We may iterate later for adding this prop later, if there is requirement.

**Comment by jurokapsiar** [comment]
will the `reactions` prop be an array of button shorthands?

**Comment by mnajdova** [comment]
@jurokapsiar I've created the Reaction component for this, but if from accessibility point of view these should be button I may change them. Let me know what you think!

**Comment by layershifter** [comment]
Was fixed in #959.
